review of docker entry point & docker documentation

return42 opened the pull request:
**WIP** .. no need to review right now .. 


------

## What does this PR do?

- review of docker-entrypoint.sh
- revision of the docker documentation

additional hints from @dalf:

* about filtron: https://github.com/searxng/searxng/blob/9a383d15334eb58e273f394f315cf46398e597bb/dockerfiles/uwsgi.ini#L35-L37
* the static files are served by uwsgi
    * uwsgi configuration: https://github.com/searxng/searxng/blob/9a383d15334eb58e273f394f315cf46398e597bb/dockerfiles/uwsgi.ini#L39-L44 (not that ETag header is not supported)
    * they are compressed when the image is built: https://github.com/searxng/searxng/blob/9a383d15334eb58e273f394f315cf46398e597bb/Dockerfile#L66-L68
    * also see https://github.com/searx/searx/pull/2242

return42 committed:
[doc] docker: revision and preparation of the documentation

- add sidebars with addition infos about commands and docker in general
- fix long lines & indentation
- correct link to https://github.com/searxng/searxng-docker

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on dockerfiles/docker-entrypoint.sh:
```diff
@@ -96,50 +103,51 @@ update_conf() {
 
 # Start uwsgi
-printf 'Listen on %s\n' "${BIND_ADDRESS}"
+echo 'Listen on %s' "${BIND_ADDRESS}"
```
it should be `echo 'Listen on' "${BIND_ADDRESS}"` without `%s` ?
or keep `printf` as above (line 124, 127, ....) ?

return42 replied:
> or keep printf as above (line 124, 127, ....) ?

I tend to used echo because I think it is more readable, but this is my very personal opinion. More experienced users say it is better to use printf ([Basically, it's a portability (and reliability) issue](https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo/65819#65819)).

on second thought, I would say it is better to stay with printf.

dalf commented on dockerfiles/docker-entrypoint.sh:
```diff
@@ -1,62 +1,67 @@
-	f)
-	    FORCE_CONF_UPDATE=1
+
+	f) FORCE_CONF_UPDATE=1 ;;
```
Is it possible to use only space and not tab in the whole file ?

return42 replied:
> Is it possible to use only space and not tab in the whole file ?

Yes, I forgot to _untabify_.

dalf commented on dockerfiles/docker-entrypoint.sh:
```diff
@@ -1,62 +1,67 @@
-export SEARX_VERSION=$(su searx -c 'python3 -c "import six; import searx.version; six.print_(searx.version.VERSION_STRING)"' 2>/dev/null)
-printf 'searx version %s\n\n' "${SEARX_VERSION}"
+BIND_ADDRESS="${BIND_ADDRESS:-${DEFAULT_BIND_ADDRESS}}"
+export BIND_ADDRESS
```
For my own knowledge why it can't be `export BIND_ADDRESS="${BIND_ADDRESS:-${DEFAULT_BIND_ADDRESS}}"` in one line ?

return42 replied:
It can, my fail.

dalf commented on dockerfiles/docker-entrypoint.sh:
```diff
@@ -1,62 +1,67 @@
+}
+
+SEARX_VERSION="$(get_searx_version)"
+export SEARX_VERSION
```
Why not `export SEARX_VERSION="$(get_searx_version)"` ?

return42 replied:
The return value of the export will always _true_ and the return value from the `get_searx_version` will be ignored, see  [SC2155: Declare and assign separately to avoid masking return values.](https://github.com/koalaman/shellcheck/wiki/SC2155)

return42 committed:
[mod] docker-entrypoint.sh - add script to make test.sh (shellcheck)

Add script docker-entrypoint.sh to shellcheck and try to _simplify_ and
_normalize_ some parts:

- fix issues reported by shellcheck
- don't mix tab and space indent
- command 'help' replaced by '-h': ./dockerfiles/docker-entrypoint.sh -h
- replace printf in help() by 'cat <<EOF'

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[brand] Dockerfile default INSTANCE_NAME=searxng

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
